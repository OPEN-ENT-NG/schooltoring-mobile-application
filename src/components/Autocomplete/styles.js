import { StyleSheet } from "react-native";

import { COLORS } from "../../styles/common";

export default StyleSheet.create({
  input: {
    backgroundColor: COLORS.WHITE
  },
  list: {
    backgroundColor: COLORS.WHITE,
    zIndex: 10,
    position: "relative"
  },
  item: {
    padding: 20
  }
});
