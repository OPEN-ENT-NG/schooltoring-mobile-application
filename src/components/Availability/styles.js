import { StyleSheet } from "react-native";

export default StyleSheet.create({
  container: {
    paddingHorizontal: 30,
    paddingVertical: 15,
    flex: 1,
    justifyContent: "space-evenly"
  },
  item: {
    flexDirection: "row",
    justifyContent: "flex-start",
    alignItems: "center",
    height: 60
  },
  itemSwitch: {
    marginRight: 20
  },
  itemText: {
    fontSize: 18
  },
  list: {
    flex: 1,
    justifyContent: "space-around"
  },
  title: {
    fontSize: 18,
    marginBottom: 40,
    flex: 0
  },
  buttonView: {
    flexDirection: "row",
    justifyContent: "flex-end",
    flex: 0,
    padding: 15
  }
});
