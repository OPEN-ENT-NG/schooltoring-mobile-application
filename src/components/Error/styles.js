import { StyleSheet } from "react-native";

const styles = StyleSheet.create({
  container: {
    flex: 1,
    flexDirection: "column"
  },
  imageCritical: {
    width: 280,
    height: 301,
    position: "absolute",
    bottom: 0,
    left: -30
  },
  imageUncritical: {
    width: 262,
    height: 301,
    position: "absolute",
    bottom: 0,
    right: -30
  }
});

export default styles;
