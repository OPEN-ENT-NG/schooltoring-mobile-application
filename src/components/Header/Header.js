import React from "react";
import { View, Text, TouchableWithoutFeedback } from "react-native";
import { StackActions } from "react-navigation";

import Icon from "react-native-vector-icons/MaterialIcons";

import getStyle from "./styles";

export default props => {
  const dispatchPop = () => {
    props.navigation.dispatch(StackActions.pop());
  };

  const styles = getStyle(props.backgroundColor);

  return (
    <View style={styles.header}>
      {(props.hasOwnProperty("noBack") && props.noBack) || (
        <View>
          <TouchableWithoutFeedback
            hitSlop={{
              top: 20,
              left: 20,
              right: 40,
              bottom: 20
            }}
            onPress={dispatchPop}
          >
            <Icon name="arrow-back" style={styles.title} />
          </TouchableWithoutFeedback>
        </View>
      )}
      <View style={styles.titleContainer}>
        {props.iconName && <Icon style={styles.title} name={props.iconName} />}
        <Text style={styles.title}>{props.title}</Text>
      </View>
    </View>
  );
};
