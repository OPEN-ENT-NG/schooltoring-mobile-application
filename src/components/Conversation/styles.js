import { StyleSheet } from "react-native";

import { COLORS } from "../../styles/common";

const textColor = {
  color: COLORS.TEXT
};

export default StyleSheet.create({
  conversation: {
    flexDirection: "row",
    paddingHorizontal: 10
  },
  avatar: {
    alignSelf: "flex-start",
    flex: 1,
    alignItems: "center",
    paddingVertical: 20,
    marginHorizontal: 5
  },
  content: {
    flex: 4,
    justifyContent: "space-evenly",
    borderBottomColor: "#000",
    borderBottomWidth: StyleSheet.hairlineWidth
  },
  name: {
    ...textColor,
    fontWeight: "bold",
    fontSize: 20
  },
  message: {
    flexDirection: "row",
    justifyContent: "space-evenly"
  },
  icon: {
    fontSize: 20,
    marginRight: 10
  },
  text: {
    ...textColor,
    flex: 5,
    fontSize: 16,
    flexWrap: "wrap"
  },
  date: {
    flexGrow: 2,
    alignItems: "flex-end"
  },
  dateText: {
    ...textColor
  }
});
