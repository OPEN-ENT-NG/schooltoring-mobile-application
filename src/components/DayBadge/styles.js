import { StyleSheet } from "react-native";

import { COLORS } from "../../styles/common";

export default StyleSheet.create({
  item: {
    justifyContent: "center",
    alignItems: "center",
    width: 30,
    height: 30,
    backgroundColor: COLORS.GREY
  },
  itemText: {
    flex: 1,
    color: COLORS.WHITE,
    fontSize: 20
  },
  available: {
    backgroundColor: COLORS.NEGATIVE
  }
});
