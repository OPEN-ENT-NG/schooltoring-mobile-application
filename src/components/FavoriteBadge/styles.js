import { StyleSheet, Platform } from "react-native";

import { COLORS } from "../../styles/common";

const shadow =
  Platform.OS === "android"
    ? {
        elevation: 4
      }
    : {
        shadowColor: COLORS.GREY,
        shadowOffset: { width: 2, height: 2 },
        shadowOpacity: 0.6,
        shadowRadius: 1
      };

export default StyleSheet.create({
  badge: {
    alignItems: "stretch",
    width: "50%",
    padding: 10
  },
  top: {
    aspectRatio: 1,
    justifyContent: "center",
    alignItems: "center",
    backgroundColor: COLORS.SECONDARY_BACKGROUND,
    ...shadow
  },
  bottom: {
    backgroundColor: COLORS.BACKGROUND,
    ...shadow,
    maxHeight: 120
  },
  name: {
    textAlign: "center",
    color: COLORS.TEXT,
    fontWeight: "bold",
    fontSize: 16
  },
  buttons: {
    flexDirection: "row",
    justifyContent: "space-around",
    alignItems: "center",
    padding: 5
  },
  chatButton: {
    width: 40,
    height: 40,
    borderRadius: 20,
    backgroundColor: COLORS.PRIMARY
  },
  loader: {
    ...StyleSheet.absoluteFillObject,
    backgroundColor: "rgba(255, 255, 255, 0.8)",
    elevation: 1000,
    zIndex: 5
  }
});
