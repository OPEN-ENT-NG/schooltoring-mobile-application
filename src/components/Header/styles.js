import { StyleSheet } from "react-native";

import { COLORS } from "../../styles/common";

export default StyleSheet.create({
  header: {
    backgroundColor: COLORS.LIGHT_GREY,
    flexDirection: "row",
    alignItems: "center",
    justifyContent: "flex-end",
    padding: 10
  },
  titleContainer: {
    flex: 0
  },
  back: { flex: 1 },
  title: {
    fontSize: 25,
    marginLeft: 10
  }
});
