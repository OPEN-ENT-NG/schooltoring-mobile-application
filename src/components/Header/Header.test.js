import React from "react";
import { Text } from "react-native";
import { shallow } from "enzyme";

import { COLORS } from "../../styles/common";
import { Header } from "./Header";

jest.mock("react-navigation", () => ({
  StackActions: {
    pop: jest.fn()
  }
}));

const callback = jest.fn();
const navigation = {
  dispatch: callback
};

describe("Render", () => {
  test("Header should render without crashing", () => {
    const rendered = shallow(<Header title="Test" navigation={navigation} />);
    expect(rendered).toMatchSnapshot();
  });

  test("An error should be logged when no props title given", () => {
    const PropTypes = jest.spyOn(global.console, "error");
    const rendered = shallow(<Header />);
    expect(PropTypes).toHaveBeenCalled();
  });
  test("Rendered title should be the same as the title prop given", () => {
    const title = "test";
    const rendered = shallow(<Header title={title} />);
    expect(
      rendered
        .find(Text)
        .first()
        .contains(title)
    ).toBeTruthy();
  });

  test("Header with noBack props set as 'false' should render a back button", () => {
    const rendered = shallow(
      <Header title="Test" noBack={false} navigation={navigation} />
    );

    const backButton = rendered.find('[id="header-back-button"]');
    const backIcon = backButton.find("Icon");

    expect(backButton.exists()).toBeTruthy();
    expect(backIcon.exists()).toBeTruthy();
  });

  test("Header with background color props should render with background color", () => {
    const rendered = shallow(
      <Header
        title="Test"
        navigation={navigation}
        backgroundColor={COLORS.PRIMARY}
      />
    );

    expect(rendered.props().style.backgroundColor).toEqual(COLORS.PRIMARY);
  });
});

describe("Interactions", () => {
  const rendered = shallow(
    <Header title="Test" noBack={false} navigation={navigation} />
  );

  test("Press on back button should trigger mocked function", () => {
    const backButton = rendered.find('[id="header-back-button"]');

    backButton.props().onPress();
    expect(navigation.dispatch).toHaveBeenCalled();
  });
});
