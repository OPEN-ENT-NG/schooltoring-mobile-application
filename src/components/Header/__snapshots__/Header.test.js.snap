// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Header should render without crashing 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#ebebeb",
      "flexDirection": "row",
      "justifyContent": "flex-end",
      "padding": 10,
    }
  }
>
  <View>
    <TouchableWithoutFeedback
      hitSlop={
        Object {
          "bottom": 20,
          "left": 20,
          "right": 40,
          "top": 20,
        }
      }
      id="header-back-button"
      onPress={[Function]}
    >
      <Icon
        allowFontScaling={false}
        name="arrow-back"
        size={12}
        style={
          Object {
            "color": "#232323",
            "fontSize": 25,
            "marginLeft": 10,
          }
        }
      />
    </TouchableWithoutFeedback>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "flex-end",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#232323",
          "fontSize": 25,
          "marginLeft": 10,
        }
      }
    >
      Test
    </Text>
  </View>
</View>
`;

exports[`Render Header with rightActions should render without crashing 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#ebebeb",
      "flexDirection": "row",
      "justifyContent": "flex-end",
      "padding": 10,
    }
  }
>
  <View>
    <TouchableWithoutFeedback
      hitSlop={
        Object {
          "bottom": 20,
          "left": 20,
          "right": 40,
          "top": 20,
        }
      }
      id="header-back-button"
      onPress={[Function]}
    >
      <Icon
        allowFontScaling={false}
        name="arrow-back"
        size={12}
        style={
          Object {
            "color": "#232323",
            "fontSize": 25,
            "marginLeft": 10,
          }
        }
      />
    </TouchableWithoutFeedback>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "flex-end",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#232323",
          "fontSize": 25,
          "marginLeft": 10,
        }
      }
    >
      Test
    </Text>
  </View>
  <PopupMenu
    onPress={[MockFunction]}
  />
</View>
`;
